src/Engine/EmitMSIL/Internals.cs

```diff
@@ -0,0 +1,235 @@
+﻿using ProtoCore.DSASM;
+using ProtoCore.Utils;
+using System;
+using System.Collections.Generic;
+using System.Reflection;
+
+namespace EmitMSIL
+{
+    public class Internals
+    {
+        private static bool ToBoolean(dynamic a)
+        {
+            switch (a)
+            {
+                case bool ab:
+                    return ab;
+
+                case long al:
+                    return al != 0;
+
+                case null:
+                    return false;
+
+                case double ad:
+                    return !Double.IsNaN(ad) && !ad.Equals(0.0);
+
+                case string str:
+                    return !string.IsNullOrEmpty(str);
+
+                case char ac:
+                    return ac != 0;
+            }
+
+            if (a is Pointer)
+            {
+                return true;
+            }
+
+            return false;
+        }
+
+        public static dynamic Add(dynamic a, dynamic b)
+        {
+            return a + b;
+        }
```

Comment:
One thing I see with this is that it could cause some differences with the old `Add` method in Dynamo. For example, in this case, an `int` can be added with a `char` and still give a valid result (the resulting ASCII value), whereas in the old VM, it would return null, just because it was implemented differently.
![image](https://user-images.githubusercontent.com/5710686/180249486-437da5f1-2f40-455b-a3ef-2d01406abdcb.png)

Reply:
@mjkkirschner trying to remember, is this approach something you tried for `GenerateRangeIL `functions, where you could define the inputs having dynamic type?

Reply:
Yes, I think many (all) of these new methods are not fully compatible in what they support and how they fail.  Something we need to look at.

Reply:
@aparajit-pratap no I didn't try dynamic, I did use `object` as my attempt at a dynamic range creation conversion, but then we decided to file followups:
1. evaluate stackvalue,object vs dynamic performance: https://jira.autodesk.com/browse/DYN-5110
2. Implement dynamic range generation for character ranges: https://jira.autodesk.com/browse/DYN-5109

where the idea was that doing 1 would give us some direction in how to implement 2. 

Maybe @sm6srw and @pinzart90 are kind of investigating 1 implicitly now though.

Reply:
Now using objects.